data/stylish-haskell.yaml

```diff
@@ -294,6 +294,83 @@ steps:
       # Default: false
       post_qualify: false
 
+      # Automatically group imports based on their module names, with
+      # a blank line separating each group. Groups are ordered in
+      # alphabetical order.
+      #
+      # By default, this groups by the first part of each module's
+      # name (Control.* will be grouped together, Data.*... etc), but
+      # this can be configured with the group_patterns setting.
+      #
+      # When enabled, this rewrites existing blank lines and groups.
+      #
+      # - true: Group imports by the first part of the module name.
+      #
+      #   > import Control.Applicative
+      #   > import Control.Monad
+      #   > import Control.Monad.MonadError
+      #   >
+      #   > import Data.Functor
+      #
+      # - false: Keep import groups as-is (still sorting and
+      #   formatting the imports within each group)
+      #
+      #   > import Control.Monad
+      #   > import Data.Functor
+      #   >
+      #   > import Control.Applicative
+      #   > import Control.Monad.MonadError
+      #
+      # Default: false
+      group_imports: false
+
+      # A list of regular expressions specifying how to group modules
+      # and how to order the groups.
+      #
+      # Each pattern defines a single group. An import belongs to the
+      # first group where the pattern matches the name of the module
+      # being imported. Matches can be partial; to require matching
+      # the entire module name, you can use ^ and $ in the pattern (eg
+      # "^pattern$").
+      #
+      # Example: ["Test\\.", "Foo\\."] will put all test imports into
```

Comment:
Maybe include some tricky examples for this case, e.g. I believe `Foo.Test.Lib` will end up in the first block but `Foo.Test` will end up in the second! Or perhaps you should put `^`s in the regexes :)

Reply:
Great call. I'll add a bit more explanation in the docs—ideally, I want to provide sufficient explanation and examples so that people could at least configure rules based on specific module name components without prior familiarity with regular expression syntax.

Reply:
I think _most_ users will want to have a leading `^`, since I expect "top-down" grouping to be the most common desire.